src/main/java/lania/Storage.java

```diff
@@ -0,0 +1,97 @@
+package lania;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+import lania.task.Deadline;
+import lania.task.Event;
+import lania.task.Task;
+import lania.task.TaskList;
+import lania.task.Todo;
+
+/**
+ * This class deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+
+    /** Represents the path of the file to be retrieved or written. */
+    private String filePath;
+
+    /**
+     * Constructor for the storage class.
+     *
+     * @param filePath The path of the file to write or read from.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * This method reads the file if it exists and stores the data
+     * into a TaskList.
+     *
+     * @return The TaskList containing the tasks specified in the file.
+     * @throws  IOException If the directory of file cannot be found.
+     */
+    public TaskList load() throws IOException {
+        Files.createDirectories(Paths.get("data/"));
+        File f = new File(filePath);
+        if (f.createNewFile()) {
+            return new TaskList();
+        } else {
+            TaskList tasks = new TaskList();
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String next = s.nextLine();
+                String[] split = next.split("\\|", 4);
+                Task t;
+                if (split[0].equals("T")) {
```

Comment:
I didn't like the raw logic being in the 4th indentation level here. Consider packaging it into a separate method?

Reply:
Thank you for the advice, abstracting the code into a method would indeed make it cleaner. Will do so!